mosql/util.py

```diff
@@ -237,29 +241,36 @@ def __repr__(self):
 star = raw('*')
 'The ``*`` keyword in SQL.'
 
-class param(str):
+class param(compat.text_type):
     '''The :func:`value` builds this type as a parameter for the prepared
     statement.
 
-    >>> value(param(''))
-    '%s'
-    >>> value(param('name'))
-    '%(name)s'
+    >>> print(value(param('')))
+    %s
+    >>> print(value(param('name')))
+    %(name)s
 
-    This is just a subclass of built-in :class:`str` type.
+    This is a subclass of the built-in text type.
 
     The :class:`___` is an alias of it.
     '''
 
     def __repr__(self):
-        return 'param(%r)' % self
+        return str('param(%r)') % super(param, self).__repr__()
 
 ___ = param
 
 # qualifier functions
 
 def _is_iterable_not_str(x):
-    return not isinstance(x, basestring) and hasattr(x, '__iter__')
+    return (not isinstance(x, compat.class_types)
+            and not isinstance(x, compat.string_types + (bytes,))
+            and hasattr(x, '__iter__'))
+
+def _coerce_str(x):
+    if isinstance(x, compat.binary_type):
+        x = x.decode('utf-8')
+    return x
```

Comment:
I am thinking ...

to keep this in Python 2, because MoSQL was a UTF-8-based project. We need to keep the basic backward compatibility.

just let binary fail in Python 3. in Python 3, you have no reason to pass a binary into here.